packages/tabs/src/TabTitle/TabTitle.tsx

```diff
@@ -14,84 +24,68 @@ import {
 } from './TabTitle.styles';
 import { BaseTabTitleProps } from './TabTitle.types';
 
-const TabTitle: ExtendableBox<BaseTabTitleProps, 'button'> = ({
-  selected = false,
-  disabled = false,
-  children,
-  className,
-  darkMode,
-  parentRef,
-  ...rest
-}: BaseTabTitleProps) => {
-  const titleRef = useRef<HTMLAnchorElement | HTMLButtonElement>(null);
-  const baseFontSize: BaseFontSize = useUpdatedBaseFontSize();
+const TabTitle = InferredPolymorphic<BaseTabTitleProps, 'button'>(
+  (
+    { children, className, disabled = false, name, onClick, ...rest },
+    fwdRef,
+  ) => {
+    const baseFontSize: BaseFontSize = useUpdatedBaseFontSize();
+    const { index, ref, id } = useDescendant(TabDescendantsContext, fwdRef);
+    const { tabPanelDescendants } = useTabPanelDescendantsContext();
+    const { as, darkMode, selectedIndex } = useTabsContext();
+    const { Component } = useInferredPolymorphic(as, rest, 'button');
 
-  const theme = darkMode ? Theme.Dark : Theme.Light;
+    const theme = darkMode ? Theme.Dark : Theme.Light;
+    const selected = index === selectedIndex;
 
-  // Checks to see if the current activeElement is a part of the same tab set
-  // as the current TabTitle. If so, and the current TabTitle is not disabled
-  // and is selected, we manually move focus to that TabTitle.
-  useEffect(() => {
-    const tabsList = Array.from(parentRef?.children ?? []);
-    const activeEl = document.activeElement;
+    const relatedTabPanel = useMemo(() => {
+      return tabPanelDescendants.find(
+        tabPanelDescendant => tabPanelDescendant.index === index,
+      );
+    }, [tabPanelDescendants, index]);
 
-    if (
-      activeEl &&
-      tabsList.indexOf(activeEl) !== -1 &&
-      !disabled &&
-      selected &&
-      titleRef.current
-    ) {
-      titleRef.current.focus();
-    }
-  }, [parentRef, disabled, selected, titleRef]);
+    const handleClick = useCallback(
+      (event: React.MouseEvent) => {
+        onClick?.(event, index);
+      },
+      [index, onClick],
+    );
 
-  const nodeText = getNodeTextContent(rest.name);
+    const nodeText = getNodeTextContent(name);
 
-  const sharedTabProps = {
-    ...rest,
-    className: cx(
-      listTitleFontSize[baseFontSize],
-      listTitleStyles,
-      listTitleModeStyles[theme].base,
-      {
-        [listTitleModeStyles[theme].selected]: selected,
-        [listTitleModeStyles[theme].hover]: !disabled && !selected,
-        [listTitleModeStyles[theme].disabled]: disabled,
-      },
-      listTitleModeStyles[theme].focus,
-      className,
-    ),
-    role: 'tab',
-    tabIndex: selected ? 0 : -1,
-    ['aria-selected']: selected,
-    name: nodeText,
-    ['data-text']: nodeText,
-    disabled,
-  } as const;
+    const componentProps = {
```

Comment:
since we only have one `Component` now, these props can be added as inline JSX. It's much easier to read & debug that way